src/fft.jl

```diff
@@ -60,6 +60,73 @@ function G_to_r(basis::PlaneWaveBasis, kpt::Kpoint, f_fourier::AbstractVector; k
 end
 
 
+@doc raw"""
+    Replace 0 by 1 in k_shift to avoid NaN in supercell fft.
```

Comment:
?

Reply:
When there is a k_shift 1/2 on one axis, the number of unit cells on this axis in the supercell is multiplied by 2.
So the dimensions of the unit cell are in that direction: `kgrid / k_shift`.

To keep the same formula when there is no k_shift I chose to replace zeros by ones in the shift.
That way I can write: `supercell_dimensions = kgrid ./ k_shift`

I wanted to avoid to treat each axis independantly, but I agree that this trick affects the readability of the code. 
If you have some ideas on how to do that without the `clean_shift` function I will be glad to change that !

Reply:
Ok I think I will just do a `supercell_size` function which will be more clear !